src/ui/public/field_editor/field_editor.js

```diff
@@ -57,6 +57,8 @@ uiModules
 
       self.cancel = redirectAway;
       self.save = function () {
+        self.saving = true;
```

Comment:
Seems reasonable, I will make it so.

I've found myself using state flags to denote asynchronous actions and their current state a lot to disable buttons, etc. (especially with React/Flux) but you're right that in this scenario the lack of a .catch could potentially cause errors.

It's also probably worthwhile, probably in another PR, to add in a .catch so if an error is thrown that we display it to the user.